### ISSUE TITLE
array of array types are not supported

### ISSUE THREAD

**Issue by ibd1279** [ibd1279/vks#8]
Somewhere between header version 273 and 278, they added 2-dimensional arrays for memory and heap types. The current header-generator doesn't know how to convert those.

**Expected behavior**

Update the vk.xml file to version 278 or newer and everything works fine.

**Actual behavior**

Update the spec file to 278 or newer and receive the following build error

```
go build
# github.com/ibd1279/vks
./vk_wrapper.go:19986:4: cannot use &x.memoryTypes (value of type *[32]_Ctype_struct_VkMemoryType) as *[32]*_Ctype_struct_VkMemoryType value in argument to func(x *[VK_MAX_MEMORY_TYPES]*_Ctype_struct_VkMemoryType) *[][]MemoryType {…}
./vk_wrapper.go:20001:4: cannot use &x.memoryHeaps (value of type *[16]_Ctype_struct_VkMemoryHeap) as *[16]*_Ctype_struct_VkMemoryHeap value in argument to func(x *[VK_MAX_MEMORY_HEAPS]*_Ctype_struct_VkMemoryHeap) *[][]MemoryHeap {…}
./vk_wrapper.go:25311:4: cannot use &x.physicalDevices (value of type *[32]_Ctype_VkPhysicalDevice) as *[32]*_Ctype_VkPhysicalDevice value in argument to func(x *[VK_MAX_DEVICE_GROUP_SIZE]*_Ctype_VkPhysicalDevice) *[][]PhysicalDevice {…}
```

**Comment by ibd1279** [comment]
While the generate is spitting out a pointer to an array of an array, the spec doesn't define it as a double array, so something is double arraying in [translate.go](https://github.com/ibd1279/vks/blob/a4dcc4367e114fd7e6877bfe248ab5cbe74fed61/header-generator/translate.go#L938).

**Comment by ibd1279** [comment]
[Line 886 ](https://github.com/ibd1279/vks/blob/a4dcc4367e114fd7e6877bfe248ab5cbe74fed61/header-generator/translate.go#L886) was assuming that a length would only appear for c-pointer list types. But for these three failing types, they are arrays inside structs, so the code was treating them as a length set array of an array, instead of treating them as a length defined inside allocated space in the struct.

Should be fixed in [33daa05](https://github.com/ibd1279/vks/commit/33daa05cc8933db2495fec564b911f66090407ac)
